### PR TITLE
[10.x] Create a method to convert base64 file to a file in the Filesystem class

### PR DESCRIPTION
**Pull request by eloquentcoder**
Description
I added a method in the file facade to convert a base64file to a file data that can be saved in a file. I also create a helper function to check if the string is a valid base64 file and it throws the following exception if it is not a valid base64 file.

Use case
It is a simple out of the box method that allows users convert their base64 images directly into a file data that they can save into a file. I have had to create methods in my laravel applications for handling this so i thought to create a method that will make this seamless and easy.

Additions:
I initially made this compatible for only base64 images but made a new pull request for other file types.

### CODE REVIEW

**Review by shaedrich — COMMENTED** [review]
You might add some tests here: [tests/Filesystem/FilesystemTest.php](https://github.com/eloquentcoder/framework/blob/10.x/tests/Filesystem/FilesystemTest.php) 😉

**Comment by eloquentcoder** [comment]
@shaedrich got it!

File: src/Illuminate/Filesystem/Filesystem.php [hunk 1]
```diff
@@ -38,7 +40,7 @@ public function exists($path)
      */
     public function missing($path)
     {
-        return ! $this->exists($path);
+        return !$this->exists($path);
```
Comment (driesvints) [inline src/Illuminate/Filesystem/Filesystem.php:43]:
```suggestion
        return ! $this->exists($path);
```

File: src/Illuminate/Filesystem/Filesystem.php [hunk 2]
```diff
@@ -331,6 +333,43 @@ public function move($path, $target)
...
+     * @return string
+     */
+    public function fromBase64(string $base64File)
+    {
+        if (! $this->isBase64File($base64File)) {
+            throw new InvalidBase64FileException();
+        }
+
+        $data_uri = Str::between($base64File, 'data:', ';');
+        if (str_contains($base64File, $data_uri)) {
```
Comment (driesvints) [inline src/Illuminate/Filesystem/Filesystem.php:349]:
```suggestion
        $data_uri = Str::between($base64File, 'data:', ';');

        if (str_contains($base64File, $data_uri)) {
```

File: src/Illuminate/Filesystem/Filesystem.php [hunk 3]
```diff
@@ -331,6 +333,44 @@ public function move($path, $target)
...
+     * @return bool
+     */
+    public function isBase64File(string $base64)
+    {
+        if (str_contains($base64, ';base64,')) {
+            $base64 = explode(';base64,', $base64)[1];
+        }
+
+        $decoded = base64_decode($base64, true);
+        return ($decoded !== false) && (strlen($decoded) > 0);
```
Comment (driesvints) [inline src/Illuminate/Filesystem/Filesystem.php:371]:
```suggestion
        $decoded = base64_decode($base64, true);

        return ($decoded !== false) && (strlen($decoded) > 0);
```

File: src/Illuminate/Filesystem/Filesystem.php [hunk 4]
```diff
@@ -358,7 +399,7 @@ public function link($target, $link)
 
         $mode = $this->isDirectory($target) ? 'J' : 'H';
 
-        exec("mklink /{$mode} ".escapeshellarg($link).' '.escapeshellarg($target));
+        exec("mklink /{$mode} ".escapeshellarg($link) . ' ' . escapeshellarg($target));
```
Comment (driesvints) [inline src/Illuminate/Filesystem/Filesystem.php:402]:
```suggestion
        exec("mklink /{$mode} ".escapeshellarg($link).' '.escapeshellarg($target));
```

File: src/Illuminate/Filesystem/Filesystem.php [hunk 5]
```diff
@@ -769,7 +810,7 @@ public function deleteDirectories($directory)
     {
         $allDirectories = $this->directories($directory);
 
-        if (! empty($allDirectories)) {
+        if (!empty($allDirectories)) {
```
Comment (driesvints) [inline src/Illuminate/Filesystem/Filesystem.php:813]:
```suggestion
        if (! empty($allDirectories)) {
```

**Comment by taylorotwell** [comment]
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If possible, please consider releasing your code as a package so that the community can still take advantage of your contributions! 

If you feel absolutely certain that this code corrects a bug in the framework, please "**@**" mention me in a follow-up comment with further explanation so that GitHub will send me a notification of your response.

**Pull request closed without merging**